File: example/c/lint-limit-business-logic.c
#include <stdio.h>

int main(void) {
  int item_total = 3000;
  int shipping = 500;

  // 仕様: 請求額は商品合計 + 送料であるべき。
  // 構文も型も正しいため、通常のLintだけでは「-」の誤りを断定できない。
  int billed_total = item_total - shipping;

  printf("請求額: %d\n", billed_total);
  return 0;
}

File: example/javascript/lint-limit-business-logic.js
const itemTotal = 3000
const shipping = 500

// 仕様: 請求額は商品合計 + 送料であるべき。
// 「-」も有効な演算なので、仕様を知らないLintは誤りと断定できない。
const billedTotal = itemTotal - shipping

console.log(`請求額: ${billedTotal}`)
